testsuite/tests/typing-layouts-iarrays/test_ignorable_product_iarray_2.ml

```diff
@@ -0,0 +1,108 @@
+(* TEST
+ include stdlib_stable;
+ include stdlib_upstream_compatible;
+ readonly_files =
+   "gen_u_iarray.ml test_gen_u_iarray.ml gen_product_iarray_helpers.ml";
+ modules = "${readonly_files}";
+ flambda2;
+ {
+   bytecode;
+ }{
+   native;
+ }{
+   flags = "-O3";
+   native;
+ }{
+   flags = "-Oclassic";
+   native;
+ }
+*)
+
+open Gen_product_iarray_helpers
+open Stdlib_stable
+open Stdlib_upstream_compatible
+
+(* If copying this test for a new product shape, you should only have to
+   change the bit between here and the next comment. See README.md in this
+   test directory. *)
+type boxed_t = float * int64
+type unboxed_t = #(float# * int64#)
```

Comment:
I'm mostly reviewing by patdiffing with the mutable array versions of these tests. I notice this one has been made a lot simpler than the corresponding mutable test, which used a deeply nested tuple here.

Is there a particular reason for this change? I do think it would be good to test nested products.